# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/moz.build
# Commit: b09ae4c3a94b
# Full Hash: b09ae4c3a94b663279189c50172c1f403d499183
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-22 21:29:07
# Regressor Bug: 1706374
# File Overlap Count: 1
# Description:
#   Bug 1706374 - Part 11: Add NodeController component bridging IPC and Ports, r=handyman
#   
#   The NodeController and NodeChannel types act as the backbone connecting the
#   existing IPC logic and driving the ports routing code. Individual NodeChannel
#   objects wrap and respond to messages from IPC::Channel, and the NodeController
# ==============================================================================

diff -r 04422175004b -r b09ae4c3a94b ipc/glue/moz.build
--- a/ipc/glue/moz.build	Mon Jun 21 21:53:10 2021 +0000
+++ b/ipc/glue/moz.build	Mon Jun 21 21:53:10 2021 +0000
@@ -10,6 +10,7 @@
 ]
 
 EXPORTS.mozilla.ipc += [
+    "AutoTransportDescriptor.h",
     "BackgroundChild.h",
     "BackgroundParent.h",
     "BackgroundUtils.h",
@@ -44,11 +45,14 @@
     "MessageLink.h",
     "MessagePump.h",
     "Neutering.h",
+    "NodeChannel.h",
+    "NodeController.h",
     "ProcessChild.h",
     "ProcessUtils.h",
     "ProtocolMessageUtils.h",
     "ProtocolUtils.h",
     "ProtocolUtilsFwd.h",
+    "ScopedPort.h",
     "ScopedXREEmbed.h",
     "SerializedStructuredCloneBuffer.h",
     "SharedMemory.h",
@@ -151,6 +155,7 @@
 ]
 
 UNIFIED_SOURCES += [
+    "AutoTransportDescriptor.cpp",
     "BackgroundImpl.cpp",
     "BackgroundUtils.cpp",
     "BrowserProcessSubThread.cpp",
@@ -172,9 +177,12 @@
     "MessageChannel.cpp",
     "MessageLink.cpp",
     "MessagePump.cpp",
+    "NodeChannel.cpp",
+    "NodeController.cpp",
     "ProcessChild.cpp",
     "ProcessUtils_common.cpp",
     "ProtocolUtils.cpp",
+    "ScopedPort.cpp",
     "ScopedXREEmbed.cpp",
     "SharedMemory.cpp",
     "Shmem.cpp",